### PR TITLE
Add per ENI DSCP setting

### PR DESCRIPTION
**Pull request by marian-pritsak**
By default, use DSCP from the original outer packet.

### CODE REVIEW

File: dash-pipeline/bmv2/dash_metadata.p4 [hunk 1]
```diff
@@ -28,11 +28,18 @@ struct conntrack_data_t {
     bool allow_out;
 }
 
+enum bit<16> dash_tunnel_dscp_mode_t {
+    INVALID = 0,
+    PRESERVE_MODEL = 1,
```
Comment (r12f) [inline dash-pipeline/bmv2/dash_metadata.p4:33]: should we make one of preserve_model or explicit set as default instead of invalid?

Comment (r12f) [reply dash-pipeline/bmv2/dash_metadata.p4:33]: also I guess you mean `PRESERVE_MODE` : D

File: dash-pipeline/bmv2/dash_metadata.p4 [hunk 2]
```diff
@@ -28,11 +28,18 @@ struct conntrack_data_t {
...
+    PIPE_MODEL = 2
```
Comment (r12f) [inline dash-pipeline/bmv2/dash_metadata.p4:34]: somehow I feel `PIPE_MODE` might not be that intuitive.... but will `EXPLICIT_MODE` be a better name? not sure if it is more clear... but no strong opinion.

File: dash-pipeline/bmv2/dash_metadata.p4 (hunk 1 above)
Comment (r12f) [reply dash-pipeline/bmv2/dash_metadata.p4:33]: Actually, please ignore my comment, I think what you are doing is right.

File: dash-pipeline/bmv2/dash_metadata.p4 (hunk 2 above)
Comment (r12f) [reply dash-pipeline/bmv2/dash_metadata.p4:34]: Please ignore my comment above. What you are doing should be right.

File: dash-pipeline/bmv2/dash_pipeline.p4 [hunk 3]
```diff
@@ -116,6 +116,8 @@ control dash_ingress(
                          bit<16> vnet_id,
                          bit<16> v4_meter_policy_id,
                          bit<16> v6_meter_policy_id,
+                         dash_tunnel_dscp_mode_t dash_tunnel_dscp_mode,
```
Comment (r12f) [inline dash-pipeline/bmv2/dash_pipeline.p4:119]: feels like we can simply call it `tunnel_dscp_mode`

Comment (r12f) [reply dash-pipeline/bmv2/dash_pipeline.p4:119]: and to make things match the dscp field can be `tunnel_dscp_value`.

**Comment by KrisNey-MSFT** [comment]
hi @marian-pritsak - is this ready for a Merge?  Let me know :)

**Pull request merged**